### ISSUE TITLE
Rule proposal: Prefer `url.href` to `url.toString`

### ISSUE THREAD

**Issue by fregante** [sindresorhus/eslint-plugin-unicorn#1655]
### Description

_Nitpick rule._

When building URLs via the `URL` constructor, casting them to string seems to be obviously done via `.toString` and `String()`, but there's actually a native property for this.

This is generally more commonly done in TypeScript since the `string` type is enforced, but many just pass the `URL` instance to `fetch` in JavaScript instead 😬

### Fail

```js
const url = new URL(location.origin);

console.log(url.toString())
console.log(String(url))
```

### Pass

```js
const url = new URL(location.origin);

console.log(url.href)
```

**Comment by fregante** [comment]
Is this applicable to other situations as well?

**Comment by sindresorhus** [comment]
Not that I can think of right now. It's fine to have a targeted rule. It makes it easier to reliably detect the correct cases.

**Comment by sindresorhus** [comment]
Any suggestions on the rule name?

**Comment by fisker** [comment]
`prefer-string-property`?

**Comment by fisker** [comment]
I'm worried how to implement this without type info. Maybe we can only check `const` and immediate use of `new URL()`.

**Comment by sindresorhus** [comment]
> Is this applicable to other classes as well?

`window.location.href` and `window.location.toString()`

**Comment by fregante** [comment]
A bit rare too 😅 

```js
document.querySelector('a').toString()
```

**Comment by fisker** [comment]
What's the replacement for `new URLSearchParams().toString()`?

**Comment by sindresorhus** [comment]
@fregante ^

**Comment by fisker** [comment]
Thought this is not equivalence, maybe `RegExp#source` over `RegExp#toString()`? 😄

**Comment by sindresorhus** [comment]
> Thought this is not equivalence, maybe RegExp#source over RegExp#toString()? 

That doesn't really make sense to me. They are used for different purposes. `.source` is used when you want to construct a regex from other regexes or modify the regex in some way. `.toString()` is used for logging/debugging.

**Comment by fregante** [comment]
> What's the replacement for `new URLSearchParams().toString()`?

Indeed there's no direct alternative. I was thinking of `url.search` instead of `url.searchParams.toString()`

**Comment by sindresorhus** [comment]
Rule name: `prefer-url-href`?

**Comment by fregante** [comment]
If there's no chance other similar situations exists, sure.

**Comment by sindresorhus** [comment]
Let's go with `prefer-url-href`. It can be made more general later on if we discover more useful cases to catch.
